src/natcap/invest/habitat_quality.py

```diff
@@ -188,281 +191,326 @@
-            args['access_vector_path'], access_raster_path, burn_values=None,
-            option_list=['ATTRIBUTE=ACCESS'])
-
-    except KeyError:
-        LOGGER.info(
-            'No Access Shape Provided, access raster filled with 1s.')
+
+    LOGGER.info('Handling Access Shape')
+    access_raster_path = os.path.join(
+        intermediate_output_dir, f'access_layer{file_suffix}.tif')
+    # create a new raster based on the raster info of current land cover
+    create_access_raster_task = task_graph.add_task(
+        pygeoprocessing.new_raster_from_base,
+        args=(cur_lulc_path, access_raster_path, gdal.GDT_Float32,
+              [_OUT_NODATA]),
+        kwargs={
+            'fill_value_list': [fill_value]
+            },
+        target_path_list=[access_raster_path],
+        dependent_task_list=[align_task],
+        task_name=f'access_raster')
+    access_task_list = [create_access_raster_task]
+
+    if 'access_vector_path' in args:
+        LOGGER.debug("Rasterize Access vector")
+        rasterize_access_task = task_graph.add_task(
+            pygeoprocessing.rasterize,
+            args=(args['access_vector_path'], access_raster_path),
+            kwargs={
+                'option_list': ['ATTRIBUTE=ACCESS'],
+                'burn_values': None
+                },
+            target_path_list=[access_raster_path],
+            dependent_task_list=[create_access_raster_task],
+            task_name=f'rasterize_access')
+        access_task_list.append(rasterize_access_task)
 
     # calculate the weight sum which is the sum of all the threats' weights
     weight_sum = 0.0
     for threat_data in threat_dict.values():
         # Sum weight of threats
-        weight_sum = weight_sum + threat_data['WEIGHT']
-
-    LOGGER.debug('lulc_path_dict : %s', lulc_path_dict)
+        weight_sum = weight_sum + threat_data['weight']
 
     # for each land cover raster provided compute habitat quality
     for lulc_key, lulc_path in lulc_path_dict.items():
-        LOGGER.info('Calculating habitat quality for landuse: %s', lulc_path)
+        LOGGER.info(f'Calculating habitat quality for landuse: {lulc_path}')
+
+        threat_convolve_task_list = []
+        sensitivity_task_list = []
 
         # Create raster of habitat based on habitat field
         habitat_raster_path = os.path.join(
-            inter_dir, 'habitat%s%s.tif' % (lulc_key, suffix))
-        map_raster_to_dict_values(
-            lulc_path, habitat_raster_path, sensitivity_dict, 'HABITAT',
-            _OUT_NODATA, values_required=False)
+            intermediate_output_dir,
+            f'habitat{lulc_key}{file_suffix}.tif')
+
+        habitat_raster_task = task_graph.add_task(
+            pygeoprocessing.reclassify_raster,
+            args=((lulc_path, 1), sensitivity_reclassify_habitat_dict,
+                  habitat_raster_path, gdal.GDT_Float32, _OUT_NODATA),
+            kwargs={
+                'values_required': False
```

Comment:
I just want to check, is it okay if there is not a sensitivity value for a given landcover code?